### PR TITLE
GEODE-10023: Fix javadocs

### PR DESCRIPTION
**Pull request by DonalEvans**
This PR is intentionally broken up into two commits.

The first commit reduces the number of errors reported in the gradle
build log due to javadocs from over 3000 to 0. It also reduces the 
number of warnings reported in the gradle build log due to javadocs
from over 4000 to 0.

After these changes, all published javadocs are now syntactically
correct, although no guarantee is made as to the content of the
documentation. Javadocs in classes that did not produce warnings or
errors in the gradle build log were not touched.

For classes in the geode-dunit, geode-junit, geode-assembly and 
geode-concurrency-test modules, if a javadoc could be converted to 
a non-javadoc comment without losing any formatting or tags, it was 
converted, since those classes are not intended for use by Geode users

The changes made were:
 - Add missing @return, @param and @throws tags
 - Fix all errors due to incorrect HTML tag formatting
 - Fix all errors due to improperly escaped characters
 - Fix all broken @link tags
 - Several minor spelling errors and typos fixed where spotted

The first commit comprises 665 files changed, 9324 insertions, and 
4053 deletions, and as such, is not realistically reviewable. Given the
benefits of the clean-up and the low impact of any possible mistakes,
a "rubber-stamp" approval is requested form code owners in this case.

In the second commit, javadocs warnings and errors will be output in
the gradle build log, and if any are present, the build will fail.

This will enable developers to know if they have broken existing
javadocs, or if any javadocs they have added are incorrect.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [N/A] Have you written or updated unit tests to verify your changes?

- [N/A] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

**Review by dschneider-pivotal — APPROVED** [review]

**Review by jinmeiliao — APPROVED** [review]

**Review by jchen21 — APPROVED** [review]
Reviewed the files that I am owner of. Looks good to me.

**Review by boglesby — APPROVED** [review]

**Review by rhoughton-pivot — APPROVED** [review]

File: geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/session/tests/ContainerInstall.java [hunk 1]
```diff
@@ -248,32 +252,36 @@ int getDefaultLocatorPort() {
...
-  /**
+  /*
    * Get the session manager class to use
    */
   public abstract String getContextSessionManagerClass();
 
   /**
    * Generates a {@link ServerContainer} from the given {@link ContainerInstall}
    *
+   * @param containerConfigHome The folder that the container configuration folder should be setup
```
Comment (gesterzhou) [inline geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/session/tests/ContainerInstall.java:280]: these 2 lines can be adjusted

Comment (DonalEvans) [reply geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/session/tests/ContainerInstall.java:280]: How do you mean?

**Review by nabarunnag — APPROVED** [review]

File: geode-assembly/build.gradle [hunk 2]
```diff
@@ -384,7 +384,7 @@ tasks.register('gfshDepsJar', Jar) {
 
 tasks.register('docs', Javadoc) {
   def docsDir = file("$buildDir/javadocs")
-  options.addStringOption('Xdoclint:none', '-quiet')
+  options.addStringOption('Xwerror', '-quiet')
```
Comment (agingade) [inline geode-assembly/build.gradle:387]: Does it break, cause failures for other devs/PRs?

Comment (DonalEvans) [reply geode-assembly/build.gradle:387]: No, the build currently passes, and will only fail if someone breaks existing javadocs or introduces new javadocs that have errors or warnings.

File: gradle/java.gradle [hunk 3]
```diff
@@ -186,7 +186,7 @@ artifacts {
 
 javadoc {
   destinationDir = file("$buildDir/javadoc")
-  options.addStringOption('Xdoclint:none', '-quiet')
+  options.addStringOption('Xwerror', '-quiet')
```
Comment (agingade) [inline gradle/java.gradle:189]: Will this be a issue with existing PRs?

**Review by agingade — APPROVED** [review]

File: gradle/java.gradle (hunk 3 above)
Comment (DonalEvans) [reply gradle/java.gradle:189]: If in-flight PRs are merged after this change and the build starts failing as a result of javadocs added by the PR being incorrect, the authors of those PRs should revert and fix the javadocs, or submit a second PR to fix the broken javadocs.

**Review by Bill — APPROVED** [review]
Bravo @DonalEvans ! Approved!

**Review by gesterzhou — APPROVED** [review]

**Review by kirklund — APPROVED** [review]

**Review by BenjaminPerryRoss — APPROVED** [review]
LGTM

**Pull request merged**